config/networking/istio-authorization-policies.yml

```diff
@@ -0,0 +1,37 @@
+#@ load("@ytt:data","data")
+
+#@ def principal(namespace, service_account):
+#@   return "cluster.local/ns/{}/sa/{}".format(namespace, service_account)
+#@ end
+
+#! https://istio.io/latest/docs/reference/config/security/authorization-policy/
+---
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: default-allow-all
+  namespace: #@ data.values.system_namespace
+spec:
+  rules:
+    - {}
+---
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: cf-api-server
+  namespace: #@ data.values.system_namespace
+spec:
+  action: DENY
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: cf-api-server
+  rules:
+    - from:
+        - source:
+            notPrincipals:
+            - #@ principal(data.values.system_namespace, "eirini-events")
```

Comment:
This limits our collaborators to pods with the `eirini-events` and `eirini-task-reporter` service accounts. This means that if those service accounts ever change we'll need to update this `AuthorizationPolicy`.

Reply:
Will `action: DENY` + `notPrincipals` set up a double negative for that rule or you see it working? just a double check :)

Reply:
I had trouble getting the `AuthorizationPolicy` to work as expected with `ALLOW`. Like it ended up only allowing traffic on the `/internal*` endpoints for the eirini principals and denying all other traffic. 😒 

So this one is set to `DENY` for everyone on the `/internal*` endpoints that is *not* in the list of principals specified. If that makes sense. I agree it's confusing.

I tested it by making the `cf-api-server` `NetworkPolicy` permissive and seeing that traffic was still denied to `/internal*` from a non-eirini Pod.

Reply:
Ah, I see, thank you!